src/components/legend/draw.js

```diff
@@ -480,24 +495,43 @@ function computeTextDimensions(g, gd) {
         height = mathjaxBB.height;
         width = mathjaxBB.width;
 
-        Drawing.setTranslate(mathjaxGroup, 0, (height / 4));
+        if(legendItem) {
+            Drawing.setTranslate(mathjaxGroup, 0, height * 0.25);
+        } else { // case of title
+            Drawing.setTranslate(mathjaxGroup, bw, height * 0.75 + bw);
+        }
     } else {
-        var text = g.select('.legendtext');
-        var textLines = svgTextUtils.lineCount(text);
-        var textNode = text.node();
+        var textEl = g.select(legendItem ?
+            '.legendtext' : '.legendtitletext'
+        );
+        var textLines = svgTextUtils.lineCount(textEl);
+        var textNode = textEl.node();
 
         height = lineHeight * textLines;
         width = textNode ? Drawing.bBox(textNode).width : 0;
 
         // approximation to height offset to center the font
         // to avoid getBoundingClientRect
-        var textY = lineHeight * (0.3 + (1 - textLines) / 2);
-        svgTextUtils.positionText(text, constants.textGap, textY);
+        var textY = lineHeight * ((textLines - 1) / 2 - 0.3);
+        if(legendItem) {
+            svgTextUtils.positionText(textEl, constants.textGap, -textY);
+        } else { // case of title
+            svgTextUtils.positionText(textEl, constants.titlePad + bw, lineHeight + bw);
+        }
     }
 
-    legendItem.lineHeight = lineHeight;
-    legendItem.height = Math.max(height, 16) + 3;
-    legendItem.width = width;
+    if(legendItem) {
+        legendItem.lineHeight = lineHeight;
+        legendItem.height = Math.max(height, 16) + 3;
+        legendItem.width = width;
+    } else { // case of title
+        if(opts.side.indexOf('left') !== -1) {
+            gd._fullLayout._legendTitleWidth = width;
+        }
+        if(opts.side.indexOf('top') !== -1) {
+            gd._fullLayout._legendTitleHeight = height;
+        }
```

Comment:
Hmm. This is incorrect when for example the legend title is longer than all the legend items:

```js
Plotly.newPlot(gd, [{
  y: [1, 2, 1]
}, {
  y: [2, 1, 2]
}], {
  legend: { 
    title: {
      text: 'My Loooooooooong Title'
    }
  }
})
```

gives:

![image](https://user-images.githubusercontent.com/6675409/69645348-07988000-1034-11ea-8826-b6982b60fbfc.png)

Reply:
... in this case the margin should be pushed to fit the legend title (i.e. the full legend `<g>`)

Reply:
Good call. Fixed in https://github.com/plotly/plotly.js/pull/4386/commits/b3765b41a744ed07a511fb34c89ce3d1f7e00c4d.